examples/linear_model/plot_quantile_regression.py

```diff
@@ -11,100 +12,296 @@
+)
+axs[0, 1].hist(y_true - y_noisy_normal, edgecolor="black")
 
-def plot_points_highlighted(x, y, model_low, model_high, ax):
-    """Plot points with highlighting."""
-    mask = y <= model_low.predict(X)
-    ax.scatter(x[mask], y[mask], c="k", marker="x")
-    mask = y > model_high.predict(X)
-    ax.scatter(x[mask], y[mask], c="k", marker="x")
-    mask = (y > model_low.predict(X)) & (y <= model_high.predict(X))
-    ax.scatter(x[mask], y[mask], c="k")
 
+axs[1, 0].plot(x, y_true, label="True generative process")
+axs[1, 0].scatter(
+    x, y_noisy_pareto, color="black", alpha=0.5, label="Noisy observations"
+)
+axs[1, 1].hist(y_true - y_noisy_pareto, edgecolor="black")
 
-fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(15, 5), sharey=True)
+axs[0, 0].set_title("Dataset with heteroscedastic Normal distributed targets")
+axs[1, 0].set_title("Dataset with asymmetric Pareto distributed target")
+axs[0, 1].set_title(
+    "Residuals distribution for heteroscedastic Normal distributed targets"
+)
+axs[1, 1].set_title(
+    "Residuals distribution for asymmetric Pareto distributed target"
+)
+axs[0, 0].legend()
+axs[1, 0].legend()
+axs[0, 0].set_ylabel("y")
+axs[1, 0].set_ylabel("y")
+axs[1, 0].set_xlabel("x")
+_ = axs[1, 1].set_xlabel("Residuals")
 
-rng = np.random.RandomState(42)
-x = np.linspace(0, 10, 100)
-X = x[:, np.newaxis]
-y = 10 + 0.5 * x + rng.normal(loc=0, scale=0.5 + 0.5 * x, size=x.shape[0])
-y_mean = 10 + 0.5 * x
-ax1.plot(x, y_mean, "k--")
+# %%
+# With the heteroscedastic Normal distributed target, we observe that the
+# variance of the noise is increasing when the value of the feature `x` is
+# increasing.
+#
+# With the asymmetric Pareto distributed target, we observe that the positive
+# residuals are bounded.
+#
+# These types of noisy targets violate one of the key assumption made when one
+# wants to use a :class:`~sklearn.linear_model.LinearRegression`: the noise is
+# expected to be normally distributed with a constant variance.
+#
+# Thus, :class:`~sklearn.linear_model.QuantileRegressor` can be used in this
+# case to estimate the conditional quantile (e.g. median) instead of the
+# conditional mean as in :class:`sklearn.linear_model.QuantileRegressor`.
+#
+# In the remainder of this tutorial, we will show how
+# :class:`~sklearn.linear_model.QuantileRegressor` can be used in practice and
+# give the intuitions regarding the fitting properties of the model. Finally,
+# we will compare the both :class:`~sklearn.linear_model.QuantileRegressor`
+# and :class:`~sklearn.linear_model.LinearRegression`.
+#
+# Fitting a `QuantileRegressor`
+# -----------------------------
+#
+# In this section, we want to estimate the conditional median as well as
+# a low and high quantiles fixed at 5% and 95%, respectively. Thus, we will get
+# three linear models, one for each quantile.
+#
+# We will use the quantile at 5% and 95% to find the training samples that
+# are outside of this interval.
```

Comment:
What do you intend to say? Something like "to find the outliers in the training sample beyond the central 90% interval"?